invokeai/app/invocations/controlnet_image_processors.py

```diff
@@ -633,3 +634,30 @@ def run_processor(self, image):
 
         processed_image = depth_anything_detector(image=image, resolution=self.resolution, offload=self.offload)
         return processed_image
+
+
+@invocation(
+    "dwpose_image_processor",
+    title="DWPose Image Processor",
+    tags=["controlnet", "dwpose", "openpose"],
+    category="controlnet",
+    version="1.0.0",
+)
+class DWPoseImageProcessorInvocation(ImageProcessorInvocation):
+    """Generates an openpose pose from an image using DWPose"""
+
+    draw_body: bool = InputField(default=True)
+    draw_face: bool = InputField(default=False)
+    draw_hands: bool = InputField(default=False)
+    image_resolution: int = InputField(default=512, ge=0, description=FieldDescriptions.image_res)
+
+    def run_processor(self, image):
```

Comment:
types pls

Reply:
types for?

Reply:
So long as when I edit it I get hints in IDE 👍

Reply:
It does.